x-pack/plugins/observability_solution/slo/public/pages/slo_details/slo_details.tsx

```diff
@@ -96,6 +99,12 @@ export function SloDetailsPage() {
     }
   }, [hasRightLicense, permissions, navigateToUrl, basePath]);
 
+  useEffect(() => {
+    if (!isLoading && slo !== undefined) {
+      onPageReady();
```

Comment:
How does this function decide how to "name" or "title" this area of the app? It's using the "pageName" value that currently has [the context update bug](https://github.com/elastic/kibana/pull/186532)? (If so, follow up question: how does that value get set?)

Reply:
This is taken care by the router context: [here](https://github.com/elastic/kibana/blob/57891ff3534781bd558b680bc511427646c774c2/packages/core/execution-context/core-execution-context-browser-internal/src/execution_context_service.ts#L124) and [there](https://github.com/elastic/kibana/blob/57891ff3534781bd558b680bc511427646c774c2/packages/shared-ux/router/impl/route.tsx#L73)

Reply:
@kdelemme How about the different tabs in the SLO details page? I was wondering if there is a way to specify in the name/title the selected tab. In this case I guess we would need to call the onPageReady when the information on the respective tab is loaded.

Reply:
I think it would be good to have a way to provide the parameters directly to the onPageReady callback. I haven't looked at the code behind so not sure if it's possible. 
In case like this, tabs are not handled by the react router and therefore does not auto map to the selected tab.

@kpatticha do you think this is something we could do?

Reply:
@kdelemme and team,  thanks for integrating the performance metric with SLOs 🥳 

> I think it would be good to have a way to provide the parameters directly to the onPageReady callback

At the moment, `onPageReady` doesn't accept any params but we are open to making changes based on use cases. In your case, how are the tabs handled? Are you referring to the SLOs details page? ex `app/slos/api-backend-java/alerts`?

Reply:
Yes, we do handle the tab from the component itself based on the tabId qs. But technically, we could decide to call the `onPageReady` when the tab component load, we just need a way to override the pageName manually

Reply:
Yeah I think I would also add that right now, it's not really clear (to me) how `onPageReady` is determining _which_ "page" is ready and how it should be labeled. Do we want the ability for every use (not just tabs) to have more semantic control over the name of the page that has been loaded / is ready, especially as we add more and more of these to to the dashboard(s)?

Reply:
What would you add? The perf tracking of individual tabs, even without a way to override the pageName right now?

Reply:
I think because the placement of the `onPageReady()` hook call is already very subjective and requires manual engineering placement, that it makes sense to also let that caller specify the page name / title that should be attached to that "ready" state. So something like `onPageReady({ pageName: 'whatever' })` would be the way all of these would work, rather than trying to derive it from some part of the app path or plugin name (I'm still not exactly sure where it pulls it from).